Fixed Export functionality issue

rameshkumarit1990 opened the pull request:
-- If a custom column data is sent to the Data Grid the values get displayed as [Object Object].
-- To Fix this issue we are passing a prop as export from the column definition.
-- This handles which columns need to be exported
-- updated package.json version

rameshkumarit1990 committed:
#0000 Added very basic related changes to grid.

rameshkumarit1990 committed:
Added very basic related changes to grid.

rameshkumarit1990 committed:
#Added changes related to very basic grid

-- pass basic as very and we get a grid with no borders and white backgrounds.

rameshkumarit1990 committed:
Added changes related to very basic grid

-- pass basic as very and we get a grid with no borders and white backgrounds.

rameshkumarit1990 committed:
#0000 Removed hide title bar prop.

rameshkumarit1990 committed:
#0000 fixed conflict in table row

rameshkumarit1990 committed:
Merge branch 'feature-redux_form_field'

rameshkumarit1990 committed:
Updated package.json version

rameshkumarit1990 committed:
Removed the hideTitleBar prop

rameshkumarit1990 committed:
Fixed Export to csv related issue and updated version

-- If a custom column data is sent to the Data Grid the values gets displayed as [Object Object].
-- To Fix this issue we are passing a prop as export from the column definition.
-- This handles which columns need to be exported
-- updated package.json version

rameshkumarit1990 committed:
Fixed conflict issue

Fixed conflict issue

rameshkumarit1990 committed:
Merge branch 'ydsood:feature-redux_form_field' into feature-redux_form_field

rameshkumarit1990 committed:
updated package.json version

updated package.json version

rameshkumarit1990 committed:
Merge branch 'feature-redux_form_field' of https://github.com/rameshkumarit1990/redux-form-datagrid into feature-redux_form_field

ydsood commented on src/components/plugins/export/ExportControls.jsx:
```diff
@@ -29,10 +31,12 @@ class ExportControls extends Component<Props> {
-          ? columnModel[j].formatter(data[i][columnModel[j].dataIndex])
-          : data[i][columnModel[j].dataIndex];
-        rowData += `${value},`;
+        if (columnModel[j].export) {
```
Where is this prop defined? I cannot find any flow definitions for this.

ydsood commented on src/components/plugins/export/ExportControls.jsx:
```diff
@@ -29,10 +31,12 @@ class ExportControls extends Component<Props> {
-        rowData += `${value},`;
+        if (columnModel[j].export) {
+          const value = columnModel[j].formatter
+            ? columnModel[j].formatter(data[i][columnModel[j].dataIndex])
```
The formatter will return a value that is displayed on the UI for this column. That can be html, a button or something with css. Not necessarily a formatted number or special field in case you are expecting that.
This breaks single responsibility principle.
We would be better off exposing an exporter function that specifically handles how the value is added to the csv being exported from the state.

rameshkumarit1990 committed:
Export issue and Added onRowClick event

-- Fixed export issue of an customized column.
-- Added a feature to handle on row click event.
-- Added styling on hover of a row if it has a row click event then the row size increases.
-- updated package.json file version

rameshkumarit1990 committed:
Revert "Export issue and Added onRowClick event"

This reverts commit f925e3567546d2583b3d621d7156b5daac51d006.

rameshkumarit1990 committed:
Revert "Revert "Export issue and Added onRowClick event""

This reverts commit 17c9697060da75db8150799f214ab45f4a0f825a.

rameshkumarit1990 committed:
Revert "Revert "Revert "Export issue and Added onRowClick event"""

This reverts commit 300f36bf003ebd91fb2f064b01748f49651657d3.

rameshkumarit1990 committed:
Fixed Export issue

-- Fixed export issue which was generating an [object object] for a customized column in excel sheet with getvalue function.
